packages/gatsby/src/utils/start-server.ts

```diff
@@ -307,6 +308,8 @@ export async function startServer(
           let serverDataPromise: Promise<IServerData> = Promise.resolve({})
 
           if (page.mode === `SSR`) {
+            clearRequireCacheRecursively(PAGE_RENDERER_PATH)
```

Comment:
this might be expensive to force actual re-import on each ssr request - any chance to clear it only after PAGE_RENDERER recompiled?

Reply:
Done in https://github.com/gatsbyjs/gatsby/pull/33398/commits/66e726926a7892b51159c926cc8f2b6ebd5a9eaa